File: Day-01-Arrays/JavaScript/268_Missing_Number.js
/*
Problem: 268. Missing Number

Pattern: Sum Formula

Time Complexity: O(n)
Space Complexity: O(1)
*/

var missingNumber = function(nums) {

    const n = nums.length;
    const expectedSum = n * (n + 1) / 2;

    let actualSum = 0;

    for (const num of nums) {
        actualSum += num;
    }

    return expectedSum - actualSum;
};

File: Day-01-Arrays/CPP/268_Missing_Number.cpp
/*
===============================================================================
                            DSA Interview Playbook
===============================================================================

Problem:
268. Missing Number

Link:
https://leetcode.com/problems/missing-number/

Pattern:
Sum Formula

Difficulty:
Easy

Time Complexity:
O(n)

Space Complexity:
O(1)

===============================================================================
*/

class Solution {
public:
    int missingNumber(vector<int>& nums) {

        int n = nums.size();
        int expectedSum = n * (n + 1) / 2;
        int actualSum = 0;

        for (int num : nums) {
            actualSum += num;
        }

        return expectedSum - actualSum;
    }
};